plat/mediatek/mt8183/drivers/gpio/mtgpio.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef MT_GPIO_H
+#define MT_GPIO_H
+
+#include <common_def.h>
+#include <stdint.h>
+/*----------------------------------------------------------------------------*/
+//  Error Code No.
+#define RSUCCESS        0
+#define ERACCESS        1
+#define ERINVAL         2
+#define ERWRAPPER       3
+/*----------------------------------------------------------------------------*/
+
+#define MAX_GPIO_PIN    (MT_GPIO_BASE_MAX)
+/******************************************************************************
+* Enumeration for GPIO pin
+******************************************************************************/
+
+typedef enum GPIO_PIN {
+	GPIO_UNSUPPORTED = -1,
+
+	GPIO0, GPIO1, GPIO2, GPIO3, GPIO4, GPIO5, GPIO6, GPIO7,
```

Comment:
nit: This seems like a pretty pointless list of enums. Why not just use raw numbers?

Reply:
We just want to standardize  the use of parameters and GPIOx seems more meaningful than 0,1,2... And ENUM type doesn't waste more binary size.